resources/User.go

```diff
@@ -0,0 +1,108 @@
+package resources
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// User structure represents user and inherits XML data and methods from Resource structure
+type User struct {
+	Resource
+}
+
+// LoginToken structure represents login token of given user with attributes token, expiration time and egid
+type LoginToken struct {
+	Token          string
+	ExpirationTime time.Time
+	EGID           int
+}
+
+// CreateUser constructs User with id
+func CreateUser(id int) *User {
+	return &User{*CreateResource("USER", id)}
+}
+
+// Password method returns password for the given user
+func (u *User) Password() (string, error) {
+	return u.Attribute("PASSWORD")
+}
+
+// AuthDriver method returns authentication driver for the given user
+func (u *User) AuthDriver() (string, error) {
+	return u.Attribute("AUTH_DRIVER")
+}
+
+// MainGroup method returns main group ID of the given user
+func (u *User) MainGroup() (int, error) {
+	return u.intAttribute("GID")
+}
+
+// Groups method returns list of groups with main group at index 0
+func (u *User) Groups() ([]int, error) {
+	elements := u.XMLData.FindElements("GROUPS/ID")
+	if len(elements) == 0 {
+		return nil, fmt.Errorf("no group")
+	}
+
+	groups := make([]int, len(elements))
+
+	for i, e := range elements {
+		id, err := strconv.Atoi(e.Text())
+		if err != nil {
+			return nil, err
+		}
+		groups[i] = id
+	}
+	return groups, nil
+}
+
+// Enabled returns true for enabled
+func (u *User) Enabled() (bool, error) {
+	ret, err := u.intAttribute("ENABLED")
+	if err != nil {
+		return false, err
+	}
+	return intToBool(ret), nil
+}
+
+// LoginTokens returns list of login tokens for the given user
+func (u *User) LoginTokens() ([]LoginToken, error) {
+	elements := u.XMLData.FindElements("LOGIN_TOKEN")
+	if len(elements) == 0 {
+		return make([]LoginToken, 0), nil
+	}
+
+	loginTokens := make([]LoginToken, len(elements))
+
+	for i, e := range elements {
+		tokenElement := e.FindElement("TOKEN")
+		if tokenElement == nil {
+			return nil, fmt.Errorf("no token in login token")
+		}
+
+		expTimeElement := e.FindElement("EXPIRATION_TIME")
+		if expTimeElement == nil {
+			return nil, fmt.Errorf("no expiration time in login token")
+		}
+
+		expTime, err := strconv.ParseInt(expTimeElement.Text(), base10, bitSize64)
+		if err != nil {
+			return nil, err
+		}
+		expirationTime := time.Unix(expTime, 0)
```

Comment:
You totally ingored what I said. When there is no expiration time, OpenNebula uses `-1` to signalize that. If you use `-1` in `time.Unix` it will return a time equal to `1969-12-31 23:59:59`.